lib/livebook/zta.ex

```diff
@@ -45,6 +45,11 @@ defmodule Livebook.ZTA do
   """
   @callback authenticate(name(), Plug.Conn.t(), keyword()) :: {Plug.Conn.t(), metadata() | nil}
 
+  @doc """
+  Logouts against the given name.
+  """
+  @callback logout(name(), Phoenix.LiveView.Socket.t()) :: :ok | :error
```

Comment:
Perhaps to avid confusion with an actual error:

```suggestion
  @callback logout(name(), Phoenix.LiveView.Socket.t()) :: :ok | :unsupported
```

Reply:
What if `@callback logout(name(), Phoenix.LiveView.Socket.t()) :: :ok | :error | :unsupported` since the request to Teams API can fail?

Reply:
So I can keep other providers with `:unsupported` and Teams can return `:ok | :error` and we don't need to pattern match the `:unsupported`